openvm/src/customize_exe.rs

```diff
@@ -328,13 +328,12 @@ impl OvmApcStats {
 
 impl<'a> Candidate<BabyBearOpenVmApcAdapter<'a>> for OpenVmApcCandidate<BabyBear, Instr<BabyBear>> {
     fn create(
-        apc: AdapterApc<BabyBearOpenVmApcAdapter<'a>>,
+        apc: Arc<AdapterApc<BabyBearOpenVmApcAdapter<'a>>>,
         pgo_program_pc_count: &HashMap<u64, u32>,
         vm_config: AdapterVmConfig<BabyBearOpenVmApcAdapter>,
         max_degree: usize,
     ) -> Self {
-        let apc_metrics =
-            get_air_metrics(Arc::new(PowdrAir::new(apc.machine().clone())), max_degree);
+        let apc_metrics = get_air_metrics(Arc::new(PowdrAir::new(apc.clone())), max_degree);
```

Comment:
This is the reason for a lot of the changes in this PR: we don't want to clone the machine here, but now we need a PowdrAir which needs an Arc<Apc>. So we pass an Arc to this method, which propagates to the `Candidate` trait, etc.